src/arrayMethodSort.js

```diff
@@ -4,8 +4,25 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 1; i < this.length; i++) {
+      const currentEl = this[i];
+      let j = i - 1;
+
+      while (
+        j >= 0 &&
+        (compareFunction
+          ? compareFunction(this[j], currentEl) > 0
+          : String(this[j]) > String(currentEl))
```

Comment:
The use of `String(this[j]) > String(currentEl)` as a fallback comparison might not be the best approach for all data types. This could lead to unexpected behavior when sorting numbers or other non-string data types. Consider using a more robust default comparison, such as comparing numbers directly if they are both numbers.